eventstore/manager.go

```diff
@@ -0,0 +1,114 @@
+package eventstore
+
+import (
+	"io"
+
+	"github.com/google/uuid"
+	ds "github.com/ipfs/go-datastore"
+	kt "github.com/ipfs/go-datastore/keytransform"
+	"github.com/ipfs/go-datastore/query"
+	"github.com/textileio/go-textile-core/threadservice"
+)
+
+var (
+	dsStoreManagerBaseKey = ds.NewKey("/manager")
+)
+
+type Manager struct {
+	io.Closer
+
+	config *StoreConfig
+
+	threadservice threadservice.Threadservice
+	stores        map[uuid.UUID]*Store
+}
+
+// NewManager hydrates stores from prefixes and start them.
+func NewManager(ts threadservice.Threadservice, opts ...StoreOption) (*Manager, error) {
+	config := &StoreConfig{}
+	for _, opt := range opts {
+		if err := opt(config); err != nil {
+			return nil, err
+		}
+	}
+
+	if config.Datastore == nil {
+		datastore, err := newDefaultDatastore(config.RepoPath)
+		if err != nil {
+			return nil, err
+		}
+		config.Datastore = datastore
+	}
+
+	// @todo: handle debug here or in store?
```

Comment:
The way it is currently, it won't cause any problems, but is a not so nice as the log level will be set for every new store. We could leverage the keytransform check utilized in Store.Close to determine if this is a managed store, or a standalone store.